library/core/src/iter/traits/iterator.rs

```diff
@@ -1591,6 +1592,160 @@ pub trait Iterator {
         Flatten::new(self)
     }
 
+    /// Calls the given function `f` for each contiguous window of size `N` over
+    /// `self` and returns an iterator over the outputs of `f`.
+    ///
+    /// In the following example, the closure is called three times with the
+    /// arguments `&['a', 'b']`, `&['b', 'c']` and `&['c', 'd']` respectively.
```

Comment:
maybe this should mention overlap, like `slice::windows()` https://doc.rust-lang.org/std/primitive.slice.html#method.windows